tokio/src/io/driver/mod.rs

```diff
@@ -292,6 +318,17 @@ impl Inner {
         source: &mut impl mio::event::Source,
         interest: mio::Interest,
     ) -> io::Result<slab::Ref<ScheduledIo>> {
+        loop {
+            match self
+                .state
+                .compare_exchange(WAITING, REGISTERING, AcqRel, Acquire)
+            {
+                Ok(_) | Err(REGISTERING) => break,
+                Err(CLOSED) => return Err(io::Error::new(io::ErrorKind::Other, "reactor gone")),
+                _ => unreachable!(),
+            }
```

Comment:
I have an uneasy feeling about this. Is it ok to spin like that here? Is there a better way to do it?